benefits/oauth/views.py

```diff
@@ -0,0 +1,44 @@
+from django.shortcuts import redirect
+from django.urls import reverse
+from django.utils.decorators import decorator_from_middleware
+
+from authlib.integrations.django_client import OAuth
+
+from benefits.core import middleware, session
+
+
+_oauth = OAuth()
+
+
+@decorator_from_middleware(middleware.AgencySessionRequired)
+@decorator_from_middleware(middleware.VerifierSessionRequired)
+def login(request):
+    verifier = session.verifier(request)
+    if not verifier.requires_authentication:
+        return redirect(session.origin(request))
+
+    oauth_client = _oauth.create_client(verifier.auth_provider.name)
+
+    if not oauth_client:
+        return redirect(session.origin(request))
+
+    redirect_uri = reverse("oauth:authorize")
+    return oauth_client.authorize_redirect(request, redirect_uri)
+
+
+@decorator_from_middleware(middleware.AgencySessionRequired)
+@decorator_from_middleware(middleware.VerifierSessionRequired)
+def authorize(request):
+    verifier = session.verifier(request)
+    if not verifier.requires_authentication:
+        return redirect(session.origin(request))
+
+    oauth_client = _oauth.create_client(verifier.auth_provider.name)
+
+    token = oauth_client.authorize_access_token(request)
```

Comment:
I'm assuming this [verifies the token](https://developers.login.gov/oidc/#token-response), though [the Authlib docs](https://docs.authlib.org/en/latest/client/frameworks.html?highlight=authorize_access_token#parsing-id-token) don't that say that explicitly.

Reply:
Agreed that it would be nice for the docs to say it more clearly. Looking through the source code, I think it does verify the token -- feel to check my reading of it:
 - `django_client`'s `OAuth` class inherits from `BaseOAuth` and is set up to use certain [classes](https://github.com/lepture/authlib/blob/master/authlib/integrations/django_client/__init__.py#L8)
 - [`_oauth.create_client`](https://github.com/lepture/authlib/blob/master/authlib/integrations/base_client/registry.py#L68) will use `DjangoOAuth2App` due to our [configuration](https://github.com/cal-itp/benefits/pull/414/files#diff-d71405aa44f1e899a9cd920e5187aa1d02affdbd0da472eca95f814b0a8c1210R166-R170)
 - [`oauth_client.authorize_access_token`](https://github.com/lepture/authlib/blob/master/authlib/integrations/django_client/apps.py#L57) will internally call the [`parse_id_token` function](https://github.com/lepture/authlib/blob/master/authlib/integrations/base_client/sync_openid.py#L32) that it gets from inheriting `OpenIDMixin`
 - `parse_id_token` [decodes](https://github.com/lepture/authlib/blob/master/authlib/integrations/base_client/sync_openid.py#L66) the `id_token` [into claims](https://github.com/lepture/authlib/blob/098bb16d7068953280bbef456bee3ff5954f19e7/authlib/jose/rfc7519/jwt.py#L86), and [calls `validate`](https://github.com/lepture/authlib/blob/master/authlib/integrations/base_client/sync_openid.py#L76) on those decoded claims
    - The implementation we'd run through is [`JWTClaims.validate`](https://github.com/lepture/authlib/blob/098bb16d7068953280bbef456bee3ff5954f19e7/authlib/jose/rfc7519/claims.py#L91)

Reply:
Hmm, as I read through `JWTClaims,validate`, it looks like it's just checking if fields are present? Not sure where to find the verification we're hoping to see.

Reply:
Here's what `token` looks like after the call to `authorize_access_token()`:

![image](https://user-images.githubusercontent.com/1783439/163048982-57302021-90d4-4979-b03d-82f01bb5493e.png)

So that contains some parsed `userinfo`.